File: compiler-group-traslator/src/test/resources/OneMillionObject3.java

public class OneMillionObject3 {
	public static void main(String[] args) {

		int i;
		OneMillionObject3A[] a;
		OneMillionObject3B b;

		a = new OneMillionObject3A[1000000];

		for (i = 0; i < 1000000; i++) {
			a[i] = new OneMillionObject3A();
			b = new OneMillionObject3B();
			a[i].b = b;
		}
	}
}

class OneMillionObject3B {

}

class OneMillionObject3A {
	public OneMillionObject3B b;
}

File: compiler-group-traslator/src/test/resources/c/OneMillionObject3.c
#include<stdio.h>
#include<stdlib.h>
#include<stdbool.h>
#include "GC.c"
#include "ExceptionHandlingFunctions.c"
VM* vm;
typedef struct OneMillionObject3B {
}OneMillionObject3B;
typedef struct OneMillionObject3A {
	struct OneMillionObject3B* b;
}OneMillionObject3A;

int main(char* args) {
	vm = newVM();
	int i;
	struct OneMillionObject3A** a;
	struct OneMillionObject3B* b;
	a = newArray(vm,"struct OneMillionObject3A*",&a ,1000000 * sizeof (struct OneMillionObject3A*), NULL);
	for (i = 0 ; i < 1000000 ; i++) {
		a[i] = (OneMillionObject3A*)newObject(vm, &a, sizeof (OneMillionObject3A));
		b = (OneMillionObject3B*)newObject(vm, &b, sizeof (OneMillionObject3B));
		(checkArray(a,i, "a", 1000000) && checkNotNull(&a[i])? a[i] : a[i])->b = b;
	}
	return 0;
}